src/Player/Player.cpp

```diff
@@ -37,7 +40,145 @@ namespace Player {
+                army_country = 3;
+            } else {
+                army_country = (countries_owned.size()) / 3; //rounded-down
+            }
+
+        /*
+         * Second, we need to consider the owned continents by the current player
+         */
+
+            //Gather continents and countries from the GameEngine
+            vector<Continent *> continents_in_map = GameEngine::GameEngine::instance()->get_map()->get_continents();
+            vector<Country *> countries_in_map = GameEngine::GameEngine::instance()->get_map()->get_countries();
+            //Declaring vectors of integers that act as counters
+            vector<int> continent_index_map (continents_in_map.size(), 0);
+            vector<int> continent_index_owned (continents_in_map.size(), 0);
+            int continents_owned = 0;
+
+            //Counting how many countries are found in each continent of the map
+            for(int a=0;  a<countries_in_map.size(); a++){
+                continent_index_map.at(countries_in_map.at(a)->get_continent_index())++;
+            }
+            //Counting how many player-owned countries are found in each continent of the map
+            for(int b=0;  b<countries_owned.size(); b++){
+                continent_index_owned.at(countries_owned.at(b)->get_continent_index())++;
+            }
+            //Therefore, if continent_index_map equals continent_index_owned at a certain index, it means that
+            //Player owns all countries in that specific continent (since a country can only be defined in a single continent).
+            for(int c=0; c<continents_in_map.size(); c++){
+                if(continent_index_map.at(c) == continent_index_owned.at(c)){
+                    //Found a continent that is owned by player
+                    continents_owned++;
+                    //Update army_continent
+                    army_continent += continents_in_map.at(c)->get_bonus();
+                }
+            }
+
+        /*
+         * Third, we consider the armies resulting from a card exchange (if applicable).
+         * If player owns more than 5 cards, then it must exchange cards
+         */
+        bool exchange_desired;
+        vector<string> print_exchange_required;
+        print_exchange_required.emplace_back("Player [" + this->get_color() + "] has more than 5 cards in their hand! Exchange required. \n Following are the cards available in their hand: ");
+        vector<string> print_hand;
+        vector<int> answer_cards;
+
+            if(this->hand->size() >= 5){
+                //Displaying warning message
+                Terminal::print(print_exchange_required);
+
+                //Displaying available cards to Player
+                for(int i=0; i<this->hand->size(); i++){
+                    print_hand.emplace_back("Card " + to_string(i) + " : Type = " + this->hand->get_cards().at(i)->getType_to_string() + ", Country: " + this->hand->get_cards().at(i)->getCountry() + ".");
```

Comment:
Would suggest making a to_string() method in cards that simply returns this form in case we need to use it somewhere else in the future

Reply:
Hand::to_string() has been implemented. Should be fixed but let's double check